tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -166,15 +166,37 @@ export function getFilteredParent(apiItem: ApiItem): ApiItem | undefined {
 }
 
 /**
- * Adjusts the name of the item as needed.
- * Accounts for method overloads by adding a suffix such as "myMethod_2".
+ * Converts invalid filename characters to underscores.
+ */
+function getFileSafeNameForName(apiItemName: string): string {
+	// eslint-disable-next-line unicorn/better-regex, no-useless-escape
+	const badFilenameCharsRegExp: RegExp = /[^a-z0-9_\-\.]/gi;
+
+	// Note: This can introduce naming collisions.
+	// TODO: once the following issue has been resolved in api-extractor, we may be able to clean this up:
+	// https://github.com/microsoft/rushstack/issues/1308
+	return apiItemName.replace(badFilenameCharsRegExp, "_").toLowerCase();
+}
+
+/**
+ * Gets a filename-safe representation of the API item's display name.
+ *
+ * @remarks
+ * - Handles invalid filename characters.
+ *
+ * - Accounts for method overloads by adding a suffix (such as "myMethod_2").
  *
  * @param apiItem - The API item for which the qualified name is being queried.
  *
  * @public
  */
-export function getQualifiedApiItemName(apiItem: ApiItem): string {
-	let qualifiedName: string = getSafeFilenameForName(apiItem.displayName);
+export function getFileSafeNameForApiItem(apiItem: ApiItem): string {
+	const unqualifiedName: string =
```

Comment:
Just do away with `unqualifiedName`? Its only use is being assigned to `qualifiedName`.

Reply:
I actually originally had it that way and changed it, haha. Might just be me being overly pedantic, but I didn't want to assign the "unqualified" name to something called "qualified".

Reply:
I think I'd have used `result` in that case 😄 .

Reply:
Fair